File: gitdot-api/src/endpoint/repository/list_trending_repositories.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::repository::RepositoryResource};

pub struct ListTrendingRepositories;

impl Endpoint for ListTrendingRepositories {
    const PATH: &'static str = "/repository/trending";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListTrendingRepositoriesRequest;
    type Response = ListTrendingRepositoriesResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListTrendingRepositoriesRequest {}

pub type ListTrendingRepositoriesResponse = Vec<RepositoryResource>;

File: gitdot-web/packages/gitdot-api/src/endpoint/repository/list-trending-repositories.ts
import { z } from "zod";
import { RepositoryResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListTrendingRepositoriesRequest = z.object({});
export type ListTrendingRepositoriesRequest = z.infer<
  typeof ListTrendingRepositoriesRequest
>;

export const ListTrendingRepositoriesResponse = z.array(RepositoryResource);
export type ListTrendingRepositoriesResponse = z.infer<
  typeof ListTrendingRepositoriesResponse
>;

export const ListTrendingRepositories = {
  path: "/repository/trending",
  method: "GET",
  request: ListTrendingRepositoriesRequest,
  response: ListTrendingRepositoriesResponse,
} as const satisfies Endpoint;
export type ListTrendingRepositories = typeof ListTrendingRepositories;
